app/models/shop.rb

```diff
@@ -11,6 +11,12 @@ def create
 
   def initialize(name = nil)
    @name = name
+    raise 'name is nil'if @name == nil
```

Comment:
the idea here is more to validate before create/update instead of when initializing.